plugins/processors/rename/rename.go

```diff
@@ -0,0 +1,101 @@
+package rename
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/processors"
+	"sync"
+)
+
+const sampleConfig = `
+  ## Measurement, tag, and field renamings are stored in separate sub-tables.
+  ## Specify one sub-table per rename operation.
+  # [[processors.rename.measurement]]
+  #   ## measurement to change
+  #   from = "kilobytes_per_second"
+  #   to = "kbps"
+
+  # [[processors.rename.tag]]
+  #   ## tag to change
+  #   from = "host"
+  #   to = "hostname"
+
+  # [[processors.rename.field]]
+  #   ## field to change
+  #   from = "lower"
+  #   to = "min"
+
+  # [[processors.rename.field]]
+  #   ## field to change
+  #   from = "upper"
+  #   to = "max"
+`
+
+type renamer struct {
+	From string
+	To   string
+}
+
+type Rename struct {
+	Measurement  []renamer
+	Tag          []renamer
+	Field        []renamer
+	measurements map[string]string
+	tags         map[string]string
+	fields       map[string]string
+	once         sync.Once
+}
+
+func (r *Rename) SampleConfig() string {
+	return sampleConfig
+}
+
+func (r *Rename) Description() string {
+	return "Rename measurements, tags, and fields that pass through this filter."
+}
+
+func (r *Rename) Apply(in ...telegraf.Metric) []telegraf.Metric {
+	r.once.Do(r.init)
+
+	for _, point := range in {
+		if newMeasurementName, ok := r.measurements[point.Name()]; ok {
+			point.SetName(newMeasurementName)
+		}
+		for oldTagName, tagValue := range point.Tags() {
```

Comment:
We would want to use TagList() here instead of Tags() because it doesn't allocate, but what might be even better is if we skipped the initialization completely and just directly used the `renamer` types:
```
for _, rule := range r.Tag {
	value, ok := point.GetTag(rule.From)
		if ok {
			point.AddTag(rule.To, value)
		}
}
```

Reply:
will do!